Snowflake.Data/Configuration/EasyLoggingConfigFinder.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.IO;
+
+namespace Snowflake.Data.Configuration
+{
+    internal class EasyLoggingConfigFinder
+    {
+        internal const string ClientConfigFileName = "sf_client_config.json";
+        internal const string ClientConfigEnvironmentName = "SF_CLIENT_CONFIG_FILE";
+        internal const string UnixHomeEnvName = "HOME";
+        internal const string WindowsHomePathExtractionTemplate = "%HOMEDRIVE%%HOMEPATH%";
+        
+        public virtual string FindConfigFilePath(string configFilePathFromConnectionString)
+        {
+            return GetFilePathFromInputParameter(configFilePathFromConnectionString)
+                    ?? GetFilePathEnvironmentVariable()
+                    ?? GetFilePathFromDriverLocation()
+                    ?? GetFilePathFromHomeDirectory()
+                    ?? GetFilePathFromTempDirectory();
+        }
+        
+        internal virtual string GetFilePathEnvironmentVariable()
```

Comment:
those `GetFile*` methods are not private only because they are mocked in the tests?